src/BudgetManager/BudgetManager.CQRS/Handlers/CategoryHandlers/GetOneCategoryHandler.cs

```diff
@@ -4,31 +4,38 @@
 using BudgetManager.Model;
 using BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation;
 using MediatR;
+using MongoDB.Driver;
 
 namespace BudgetManager.CQRS.Handlers.CategoryHandlers
 {
-    public class GetOneCategoryHandler: IRequestHandler<GetOneCategoryQuery, CategoryResponse>
+    public class GetOneCategoryHandler : IRequestHandler<GetOneCategoryQuery, CategoryResponse>
     {
-        private readonly IBaseRepository<Category> _categoryRepository;
+        private readonly IBaseRepository<User> _userRepository;
         private readonly IMapper _mapper;
 
-        public GetOneCategoryHandler(IBaseRepository<Category> categoryRepository, IMapper mapper)
+        public GetOneCategoryHandler(IBaseRepository<User> userRepository, IMapper mapper)
         {
-            _categoryRepository = categoryRepository;
+            _userRepository = userRepository;
             _mapper = mapper;
         }
 
         public async Task<CategoryResponse> Handle(GetOneCategoryQuery request, CancellationToken cancellationToken)
         {
-            var id = request.id;
+            var userId = request.queryDto.UserId;
+            var categoryId = request.queryDto.Id;
 
-            var category = await _categoryRepository.FindByIdAsync(id, cancellationToken);
-            if (category == null)
-            {
-                return null;
-            }
+            var filter = Builders<User>.Filter.Eq(u => u.Id, userId)
+                & Builders<User>.Filter.ElemMatch(u => u.Categories, c => c.Id == categoryId);
+            var projection = Builders<User>.Projection.Include(u => u.Categories)
+                .ElemMatch(u => u.Categories, c => c.Id == categoryId);
 
-            var mappedCategory = _mapper.Map<CategoryResponse>(category);
+            var response = await _userRepository.FilterBy<User>(filter, projection, cancellationToken);
+            var listOfUsers = response.ToList();
+
+            if (response is null || listOfUsers is null || listOfUsers.Count < 1) { return null; }
+
+            var usersCategory = listOfUsers[0].Categories.FirstOrDefault();
```

Comment:
I think a better way to do this is, instead of createing listOfUsers, create var user = response.FirstOrDefault(), then check if user == null